screeps-game-api/src/constants.rs

```diff
@@ -513,7 +517,7 @@ pub const STORAGE_HITS: u32 = 10_000;
 
 /// Translates `STRUCTURE_*` constants.
 #[repr(u32)]
-#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, FromPrimitive, Serialize, Deserialize)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
```

Comment:
I'm not sure I understand the difference between `Exit` which uses `FromPrimitive` and `StructureType` which doesn't. They're both integers. Unless you want to be pedantic and say that `u32` are natural numbers and not integers 😛

Reply:
Main difference is that in screeps `StructureType` is actually represented by strings like `"spawn"`, `"extension"`, `"road"`, etc. and the numbering code we use is entirely internal to `screeps-game-api`. I mean it uses `__structure_type_num_to_str` from `javascript/utils.js` to translate integers to the actual constant names.

I was thinking it'd be nice to have `FromPrimitive` for screeps constants which actually are integer constants, like Exit, Direction, ReturnCode, Find*, Color since users might actually encounter those? But it's mostly contrived reasoning to find some way to make it reasonable to have `FromPrimitive` on `ReturnCode` in particular since we use that.